### PR TITLE
New article about changing password URL

### LINKED ISSUE #3793: content: Change Password URL [2020-08-31]
* [draft](https://docs.google.com/document/d/16kpLuQfInXSfn_Bvr_7rOh8ZZWsGQA-WL8dDdXSgCVc/edit#heading=h.het8m3niurcm)
* [thread](https://groups.google.com/a/google.com/g/webdev-content/c/Tz-c8bvjr80)
* next step: @agektmr to finish drafting the content

### PR DESCRIPTION
**Pull request by agektmr**
A new article about changing password URL

Fixes #3793

### CODE REVIEW

File: src/site/content/en/blog/change-password-url/index.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+Set a redirect from `/.well-known/change-password` to the change password page
+of your website. This will enable password managers to navigate your users
+directly to that page.
+
+## Introduction
+
+As you may know, [passwords are not the best way to manage
+accounts](https://security.googleblog.com/2017/11/new-research-understanding-root-cause.html).
+Luckily, there are emerging technologies such as
+[WebAuthn](https://webauthn.io/) and techniques such as one time passwords that
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:28]:
```suggestion
[WebAuthn](https://webauthn.io/) and techniques such as one-time passwords that
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+Generating and autofilling passwords using a password manager have already
+served the web well, but considering their lifecycle, updating the passwords
+whenever it's required is as important as generating and autofilling. To
+properly leverage that, password managers are adding a new feature:
+
+**Detect vulnerable passwords and suggest updating them**: Password managers can
+detect passwords that are reused, analyze the entropy and weakness of them, and
+even detect potentially leaked passwords or ones that are known to be unsafe
+from sources such as [https://haveibeenpwned.com/](https://haveibeenpwned.com/).
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:63]:
```suggestion
from sources such as [Have I Been Pwned](https://haveibeenpwned.com/).
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+Thanks to Apple Safari's
+[implementation](https://developer.apple.com/documentation/safari-release-notes/safari-13-release-notes),
+`/.well-known/change-password`, has already been available on some major
+websites for a while:
+
+* [Google](https://accounts.google.com/.well-known/change-password)
+* [GitHub](https://github.com/.well-known/change-password)  
+* [Facebook](https://www.facebook.com/.well-known/change-password)
+* [Twitter](http://twitter.com/.well-known/change-password)
+* [Wordpress](https://wordpress.com/.well-known/change-password)
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:133]:
```suggestion
* [WordPress](https://wordpress.com/.well-known/change-password)
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+Try them yourself and do the same for yours!
+
+### Browser compatibility
+
+A well-known URL for changing passwords has been [supported in Safari since
+2019](https://webkit.org/blog/9170/safari-technology-preview-84-with-safari-13-features-is-now-available/).
+Chrome's password manager is starting to support it from version 86 onwards and
+other Chromium-based browsers may follow. [Firefox considers it worth
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:142]:
```suggestion
Chrome's password manager is starting to support it from version 86 onwards
(which is scheduled for Stable release in late October 2020)
and other Chromium-based browsers may follow. [Firefox considers it worth
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+implementing](https://mozilla.github.io/standards-positions/#change-password-url),
+but has not signalled that it plans to do so as of August 2020.
+
+### Chrome's password manager behavior
+
+Let's have a look at how Chrome's password manager treats vulnerable passwords.
+
+Chrome's password manager is able to check for leaked passwords. By navigating
+to `chrome://settings/passwords` users can run Check passwords against stored
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:151]:
```suggestion
to `chrome://settings/passwords` users can run **Check passwords** against stored
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+passwords, and see a list of passwords that are recommended for update.
+
+<figure class="w-figure">
+  <img src="./check-passwords.png" width="1478" height="845" />
+  <figcaption class="w-figcaption">
+    Chrome's "check passwords" functionality
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:157]:
```suggestion
    Chrome's <b>Check passwords</b> functionality
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+  </figcaption>
+</figure>
+
+By clicking the Change password button next to a password that is recommended to
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:161]:
```suggestion
By clicking the <b>Change password</b> button next to a password that is recommended to
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+### Chrome's password manager behavior
+
+Let's have a look at how Chrome's password manager treats vulnerable passwords.
+
+Chrome's password manager is able to check for leaked passwords. By navigating
+to `chrome://settings/passwords` users can run Check passwords against stored
+passwords, and see a list of passwords that are recommended for update.
+
+<figure class="w-figure">
+  <img src="./check-passwords.png" width="1478" height="845" />
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:155]:
```suggestion
  <img src="./check-passwords.png" width="1478" height="845" class="w-screenshot"/>
```

File: src/site/content/en/blog/change-password-url/index.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+When you are deploying `/.well-known/change-password`, make sure that your
+server returns `404 Not Found` for any non-existing contents.
+
+{% endDetails %}
+
+## Resources
+
+If you have any feedback on the specification, please file an issue to [the spec
+repository](https://github.com/wicg/change-password-url/issues).
+
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:209]:
```suggestion
## Feedback

If you have any feedback on the specification, please file an issue to [the spec
repository](https://github.com/wicg/change-password-url/issues).

## Resources

```

File: src/site/content/en/blog/change-password-url/index.md [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+hero: hero.jpg
+alt: A drawing of a woman using OTP to log in to a web app.
+
+description: |
+  By redirecting requests to /.well-known/change-password to the change password URL, you can let users update their passwords easier than before.
+
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - identity
+  - security
```
Comment (kaycebasques) [inline src/site/content/en/blog/change-password-url/index.md:16]:
```suggestion
title: Help users change passwords easily by adding a well-known URL for changing passwords
subhead: Redirect a request to `/.well-known/change-password` to the change-passwords URL
authors:
  - agektmr
date: 2020-08-31
hero: hero.jpg
alt: Neon lights shaped into various symbols that are commonly used in passwords.
description: |
  By redirecting requests to /.well-known/change-password to the change password URL, you can let users update their passwords easier than before.
tags:
  - blog # blog is a required tag for the article to show up in the blog.
  - identity
  - security
```

**Comment by kaycebasques** [comment]
@agektmr I'm not sure if our scheduled publish feature is working anymore, so I'll leave it up to you to merge this when it's ready

Also, I linked to this guide from https://web.dev/secure/#prevent-info-leaks --- feel free to remove it if that's not correct (here's a [link](https://github.com/GoogleChrome/web.dev/blob/change-password-url/src/site/_data/paths/secure.json#L22) that will enable you to quickly edit from the GitHub UI) but please make sure the CI stuff passes before you merge

**Comment by agektmr** [comment]
@kaycebasques Thanks Kayce, but this article is more for /identity rather than /secure . I'll remove it from /secure .

**Pull request merged**